File: Python/MoveZeroes.py
def move_zeroes(nums):

    non_zero_idx = 0

    for i in range(len(nums)):

        if nums[i] != 0:

            nums[non_zero_idx] = nums[i]

            non_zero_idx += 1

    for i in range(non_zero_idx, len(nums)):

        nums[i] = 0

    print(nums)

nums = list(map(int, input().split()))

move_zeroes(nums)

File: Java/MoveZeroes.java
package com.example;

import java.util.Scanner;
import java.util.Arrays;

public class MoveZeroes {

	public static void main(String[] args) {

		Scanner sc= new Scanner(System.in);
		
		int n = sc.nextInt();
		
		int[] nums = new int[n];

		for(int i = 0; i < n; i ++)
		{
			nums[i] = sc.nextInt();
		}
		
		moveZeroes(nums);
	}
	
	public static void moveZeroes(int[] nums)
	{
		int nonZeroIdx = 0;
		
		for(int i = 0; i < nums.length; i ++)
		{
			if(nums[i] != 0)
			{
				nums[nonZeroIdx ++] = nums[i];
			}
		}
		
		while(nonZeroIdx < nums.length)
		{
			nums[nonZeroIdx ++] = 0;
		}
		
		System.out.println(Arrays.toString(nums));
	}

}
